sdk/ai/azure-ai-projects/azure/ai/projects/telemetry/agents/_ai_agents_instrumentor.py

```diff
@@ -0,0 +1,1369 @@
+from azure.ai.projects.models._patch import AgentEventHandler, ToolSet
+
+_Unset: Any = object()
+
+try:
+    # pylint: disable = no-name-in-module
+    from azure.core.tracing import AbstractSpan, SpanKind  # type: ignore
+    from opentelemetry.trace import StatusCode, Span
+
+    _tracing_library_available = True
+except ModuleNotFoundError:
+
+    _tracing_library_available = False
+
+
+__all__ = [
+    "AIAgentsInstrumentor",
+]
+
+
+_agents_traces_enabled: bool = False
+_trace_agents_content: bool = False
+
+
+class TraceType(str, Enum, metaclass=CaseInsensitiveEnumMeta):  # pylint: disable=C4747
+    """An enumeration class to represent different types of traces."""
+
+    AGENTS = "Agents"
+
+
+class AIAgentsInstrumentor:
+    """
+    A class for managing the trace instrumentation of AI Agents.
+
+    This class allows enabling or disabling tracing for AI Agents.
+    and provides functionality to check whether instrumentation is active.
+
+    """
+
+    def __init__(self):
+        if not _tracing_library_available:
+            raise ModuleNotFoundError(
+                "Azure Core Tracing Opentelemetry is not installed. "
+                "Please install it using 'pip install azure-core-tracing-opentelemetry'"
+            )
+        # In the future we could support different versions from the same library
+        # and have a parameter that specifies the version to use.
+        self._impl = _AIAgentsInstrumentorPreview()
+
+    def instrument(self) -> None:
+        """
+        Enable trace instrumentation for AI Agents.
+
+        """
+        self._impl.instrument()
+
+    def uninstrument(self) -> None:
+        """
+        Remove trace instrumentation for AI Agents.
+
+        This method removes any active instrumentation, stopping the tracing
+        of AI Agents.
+        """
+        self._impl.uninstrument()
+
+    def is_instrumented(self) -> bool:
+        """
+        Check if trace instrumentation for AI Agents is currently enabled.
+
+        :return: True if instrumentation is active, False otherwise.
+        :rtype: bool
+        """
+        return self._impl.is_instrumented()
+
+
+class _AIAgentsInstrumentorPreview:
+    """
+    A class for managing the trace instrumentation of AI Agents.
+
+    This class allows enabling or disabling tracing for AI Agents.
+    and provides functionality to check whether instrumentation is active.
+    """
+
+    def _str_to_bool(self, s):
+        if s is None:
+            return False
+        return str(s).lower() == "true"
+
+    def instrument(self):
+        """
+        Enable trace instrumentation for AI Agents.
+
+        This method checks the environment variable
+        'AZURE_TRACING_GEN_AI_CONTENT_RECORDING_ENABLED' to determine
+        whether to enable content tracing.
+        """
+        if not self.is_instrumented():
+            var_value = os.environ.get("AZURE_TRACING_GEN_AI_CONTENT_RECORDING_ENABLED")
```

Comment:
Everything that you can control with an environment variable should have a corresponding API entry point.